src/components/Avatar/AvatarGroup/AvatarGroup.tsx

```diff
@@ -0,0 +1,82 @@
+import { ComponentElement, ComponentState } from "react"
+import { Avatar, AvatarProps } from ".."
+import { AvatarBase } from "../AvatarBase"
+import styles from "./AvatarGroup.module.css"
+
+type AvatarGroupChildren =
+  | ComponentElement<AvatarProps, ComponentState>
+  | ComponentElement<AvatarProps, ComponentState>[]
+
+export interface AvatarGroupProps {
+  max?: number
+  size?: AvatarProps["size"]
+  renderHiddenCount?: (hiddenCount: number) => JSX.Element
+  children: AvatarGroupChildren
+}
+
+export function AvatarGroup({
+  size = "sm",
+  max = 3,
+  children,
+  renderHiddenCount,
+}: AvatarGroupProps) {
+  const childrenArray = Array.isArray(children) ? children : [children]
+  const slicedChildren = childrenArray.slice(0, max)
+  const hiddenCount = childrenArray.length - slicedChildren.length
+
+  if (childrenArray.some((child) => child.type.name !== "Avatar")) {
+    throw new Error("<AvatarGroup /> only accepts <Avatar /> children")
+  }
```

Comment:
Is this a given with the typing of `AvatarGroupChildren`?

Reply:
So yes it is, however you don't get the type inference alone and someone could abuse the api very easily and just pass in whatever they wanted.